lib/internal/Magento/Framework/TranslateInterface.php

```diff
@@ -52,4 +52,12 @@ public function setLocale($locale);
      * @return string
      */
     public function getTheme();
+
+    /**
+     * Set theme
+     *
+     * @param string $theme
+     * @return \Magento\Framework\TranslateInterface
+     */
+    public function setTheme($theme);
```

Comment:
It's a bad practice to introduce public methods for using them in tests. Please remove it

Reply:
Look. You just need to mock only newly injected object and mock their methods to make sure they return what you expect. In that case when you execute `applyCustomDesign` method no other objwects will be called except for the testing one `Design `. Just a gently reminder: Unit tests checks objects without running its dependencies, i.e. in isolation.